pkg/controllers/allocation/filter.go

```diff
@@ -39,7 +39,8 @@ func (f *Filter) GetProvisionablePods(ctx context.Context, provisioner *v1alpha4
 	}
 	// 2. Filter pods that aren't provisionable
 	provisionable := []*v1.Pod{}
-	for _, p := range pods.Items {
+	for i := range pods.Items {
```

Comment:
curious why we'd switch to using `i` here rather than just  ranging over the actual pod?  We do the range over the object ignoring the index in other loops.

Reply:
Probably to avoid variations of https://dev.to/kkentzo/the-golang-for-loop-gotcha-1n35

Reply:
@bwagner5 the difference is that in other loops, the address of the range object isn't being taken (and there's no closure going on either), so the linter isn't warning about those.